pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -147,6 +235,18 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
 	} else if len(murs) == 1 {
 		mur := murs[0]
 
+		// If the user has been banned, then we need to delete the MUR
+		if banned {
+			err = r.client.Delete(context.TODO(), &mur)
+			if err != nil {
+				return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, instance, r.setStatusFailedToDeleteMUR, err,
+					"Error deleting MasterUserRecord for banned UserSignup")
+			}
+
+			reqLogger.Info("Deleted MasterUserRecord", "Name", mur.Name)
+			return reconcile.Result{}, r.updateStatus(reqLogger, instance, r.setStatusBanning)
+		}
```

Comment:
It's almost the same as the following (`if instance.Spec.Deactivated`) block.
WDYT about unifing them?
Something like this:
```
if banned {
    return r.deleteMur(mur, r.setStatusBanning, r.setStatusFailedToBan)
}
if deactivated {
    return r.deleteMur(mur, r.setStatusBanning, r.setStatusFailedToBan)
}
...
func (r *ReconcileUserSignup) deleteMur(mur, inProgressStatusUpdater, failedStatusUpdater func(...)) (reconcile.Result, error) {
	if err := r.updateStatus(reqLogger, instance, inProgressStatusUpdater); err != nil {
		logError...
		return reconcile.Result{}, err
	}
	if err := r.client.Delete(context.TODO(), &mur); err != nil {
		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, instance, r.failedStatusUpdater, err,
					"Error deleting MasterUserRecord")
	}

	reqLogger.Info("Deleted MasterUserRecord", "Name", mur.Name)
	return reconcile.Result{}, nil
}
```

Reply:
Good idea, I've created a unified function as suggested.